lua/tint.lua

```diff
@@ -169,14 +169,71 @@ local function setup_namespaces()
     __.tint_ns = vim.api.nvim_create_namespace("_tint_dim")
   end
 
-  for hl_group_name, hl_def in pairs(get_global_highlights()) do
+  for hl_group_name, hl_def in pairs(get_global_highlights(0)) do
     -- Ensure we only have valid keys copied over
     hl_def = ensure_valid_hl_keys(hl_def)
     set_default_ns(hl_group_name, hl_def)
-    set_tint_ns(hl_group_name, hl_def)
+    set_tint_ns(hl_group_name, hl_def, __.tint_ns)
   end
 end
 
+local function add_namespace(ns_id, suffix)
+  __["tint_ns_" .. suffix] = vim.api.nvim_create_namespace("_tint_dim_" .. suffix)
+
+  for hl_group_name, hl_def in pairs(get_global_highlights(ns_id)) do
+    -- Ensure we only have valid keys copied over
+    hl_def = ensure_valid_hl_keys(hl_def)
+    set_tint_ns(hl_group_name, hl_def, __["tint_ns_" .. suffix])
+  end
+end
+
+---@param winid number
+---@return number
+local function get_untint_ns_id(winid)
+  local untint_ns_id = vim.w[winid].untint_ns_id
+  if untint_ns_id == nil then
+    untint_ns_id = vim.api.nvim_get_hl_ns and vim.api.nvim_get_hl_ns({ winid = winid })
+    if untint_ns_id == nil or untint_ns_id < 0 then
+      untint_ns_id = 0
+    end
+    vim.api.nvim_win_set_var(winid, "untint_ns_id", untint_ns_id)
+  end
+  return untint_ns_id
+end
+
+---@param winid number
+---@return number
+local function get_tint_ns_id(winid)
+  local untint_ns_id = get_untint_ns_id(winid)
+
+  local tint_ns_id
+  if untint_ns_id == 0 then
+    tint_ns_id = __.tint_ns
+  else
+    local ns_suffix
+    for ns_name, ns_id in pairs(vim.api.nvim_get_namespaces()) do
```

Comment:
Is this necessary? The docs for `nvim_create_namespace` say:

```
    Namespaces can be named or anonymous. If `name` matches an existing
    namespace, the associated id is returned. If `name` is an empty string a
```

which if the goal here is to try to find an existing namespace or not, we can likely just call this function to "create or get", right? I have not gone through the code thoroughly yet so my interpretation of what you are trying to do here might be wrong.

Reply:
not necessary for the changes to work. 

While debugging I found a lot of hl-namespaces, and all had pretty clear namespace names, so everyone can figure out where they are coming from. Thats why I decided to go with a specific name depending on found colorscheme :)